### ISSUE TITLE
Accessibility: Dropdown not reading country options

### ISSUE THREAD

**Issue by okkays** [jackocnr/intl-tel-input#1536]
👋 thanks for this great plugin - let me know if you need more info / help reproducing!

### Steps to reproduce

1. Enable a screenreader (we've tried on JAWS and Mac's VoiceOver)
2. Tab to select the country box
3. Press enter to expand the country box
4. Use the down arrow keys to highlight country choices

### Expected behaviour

When using arrows keys to navigate the box, options should be read aloud, so that visually impaired users know what option is highlighted.

Selecting a choice with Enter also doesn't read the country name or code.

### Actual behaviour

The screen reader doesn't read the highlighted country

![accessibility](https://github.com/jackocnr/intl-tel-input/assets/15126660/2e0a180f-bdce-44b0-bf84-742638b50eac)

I _do_ see that the combobox roles are being populated (thank you for putting in the effort to add those!) so I'm not sure why it's not behaving as expected.

### Initialisation options

I'm able to reproduce this against the demo on https://intl-tel-input.com/

**Comment by jackocnr** [comment]
Thanks for raising this issue. I'm afraid I don't have time to work on this myself right now, but would welcome a pull request.

**Comment by jackocnr** [comment]
Note: we used to follow the ARIA rules of this [combobox example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/), but now I'm guessing we need to switch to a [Combobox with Autocomplete example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/) (when countrySearch is enabled)

**Comment by jackocnr** [comment]
I've had a stab at this in v20.0.0 - please give it a try and let me know your thoughts. I'll close this for now, but we can re-open if need be.
